src/javascript/app/pages/trade/digit_ticker.js

```diff
@@ -0,0 +1,242 @@
+    const calculateOffset = () => {
+        const left_offset = calculateLeftMargin(el_container.offsetWidth);
+        return (+current_spot * digit_block_size) + left_offset - style_offset_correction;
+    };
+
+    const markAsLost = () => {
+        if (!el_peek_box || !el_peek) {
+            setElements();
+        }
+        el_peek.classList.remove('digit-winning', 'digit-running');
+        el_peek_box.classList.remove('digit-winning', 'digit-running');
+        el_peek.classList.add('digit-losing');
+        el_peek_box.classList.add('digit-losing');
+    };
+
+    const markDigitAsWon = (digit) => {
+        el_container.querySelector(`.digit-${digit}`).classList.add('digit-won');
+    };
+
+    const markDigitAsLost = (digit) => {
+        el_container.querySelector(`.digit-${digit}`).classList.add('digit-lost');
+    };
+
+    const markAsWon = () => {
+        if (!el_peek_box || !el_peek) {
+            setElements();
+        }
+        el_peek_box.classList.remove('digit-losing', 'digit-running');
+        el_peek_box.classList.add('digit-winning');
+        el_peek.classList.remove('digit-losing', 'digit-running');
+        el_peek.classList.add('digit-winning');
+    };
+
+    const setElements = () => {
+        el_peek     = el_container.querySelector('.peek');
+        el_peek_box = el_container.querySelector('.peek-box');
+        el_mask     = el_peek_box.querySelector('.peek-box > .mask');
+    };
+
+    const update = (current_tick_count, { quote, epoch }) => {
+        setElements(epoch);
+        el_container.classList.remove('invisible');
+        adjustBoxSizes();
+        current_spot = quote.substr(-1);
+
+        el_mask.innerText = `${current_tick_count} / ${total_tick_count}`;
+
+        el_peek_box.classList.add('digit-running');
+        el_peek.classList.add('digit-running');
+
+        el_peek_box.setAttribute('style', `transform: translateX(${calculateOffset()}px)`);
+    };
+
+    const remove = () => {
+        window.onresize = null;
+    };
+
+    const countDecimals = function(value) {
+        if (Math.floor(value) !== value) return value.toString().split('.')[1].length || 0;
+        return 0;
+    };
+
+    const calculateDistance = (old_digit, new_digit) => Math.abs(old_digit - new_digit);
+
+    const countUp = (start, end, duration, element, render) => {
+        const decimal_points = countDecimals(start);
+        const range = calculateDistance(start, end);
+        const step = Math.abs(range / 60);
+        const increment = end > start ? (1 * step) : (-1 * step);
+        let current = start;
+        let i = 0;
+
+        const renderTick = () => {
+            i++;
+            current += increment;
+
+            element.innerHTML = render(current.toFixed(decimal_points));
+
+            if (i <= 15) {
+                requestAnimationFrame(renderTick);
+            } else {
+                current = end;
+                element.innerHTML = render(current.toFixed(decimal_points));
```

Comment:
do we need to call cancelAnimationFrame here?

Reply:
I don't think we need to call it, since we are calling this function 16 times and then bail.